cluster-operator/src/main/java/io/strimzi/operator/cluster/model/Utils.java

```diff
@@ -0,0 +1,117 @@
+        try {
+            if (data.get(key) == null) { // value is not set in config map -> will be used default value
+                return defaultValue;
+            }
+            return Integer.parseInt(data.get(key));
+        } catch (NumberFormatException e)  {
+            String msg = " is corrupted";
+            if (data.get(key).isEmpty()) {
+                msg = " is empty";
+            }
+            throw new InvalidConfigMapException(key, msg);
+        }
+    }
+
+    public static boolean getBoolean(Map<String, String> data, String key, boolean defaultValue) {
+        if (data.get(key) == null) { // value is not set in config map
+            return defaultValue;
+        } // java parses "truue" as false. We need to be more strict
+        if (data.get(key).equals("true")) {
+            return true;
+        } else if (data.get(key).equals("false")) {
+            return false;
+        } else {
+            throw new InvalidConfigMapException(key, " is corrupted");
+        }
+    }
+
+    public static Storage getStorage(Map<String, String> data, String key) {
+        try {
+            if (data.get(key) == null) { // value is not set in config map
+                return new Storage(Storage.StorageType.PERSISTENT_CLAIM);
+            }
+            return Storage.fromJson(new JsonObject(data.get(key)));
+        } catch (Exception e) {
+            throw new InvalidConfigMapException(key, " is corrupted");
+        }
+    }
+
+    public static String getNonemptyString(Map<String, String> data, String key, String defaultValue) {
+        if (data.get(key) == null) { // value is not set in config map -> will be used default value
+            return defaultValue;
+        }
+        if (data.get(key).length() == 0) {
+            throw new InvalidConfigMapException(key, " is empty");
+        }
+        return data.get(key);
+    }
+
+    public static String getJsonCorruptionBlame(DecodeException de, String config) {
+        String token = "";
+        token = de.getMessage();
+        if (token.contains("Illegal unquoted character")) {
+            return "JSON quotation";
+        }
+        if (token.contains("Failed to decode: Unexpected end-of-input")) {
+            return "JSON braces";
+        }
+        token = token.replace("Failed to decode: Unrecognized token '", "");
+        token = token.substring(0, token.indexOf("'"));
+
+        String[] entries = config.replace("{", "").replace("}", "")
+               .replace("\"", "").replace(" ", "")
+                .replace(",", "").split("\n");
+        int i;
+        String blame = "";
+        for (i = 0; i < entries.length; ++i) {
+            if (entries[i].split(":")[1].equals(token)) {
+                blame = entries[i].split(":")[0];
+                break;
+            }
+        }
+        return blame;
+    }
+
+    public static <T extends AbstractConfiguration> T getConfig(Map<String, String> data, String key) {
```

Comment:
We should use either "config" or "configuration" consistently. 

And TBH I don't think this method achieves much: I think it would be better either with separate `getKafkaConfuration()`, `getZookeeperConfuration()` and `getKafkaConnectConfuration()` methods, avoiding the `if/else if`. Or, alternatively, you could just have the current callers of this method call `getJson()` directly and instantiate the relevant type of `*Configuration` themselves.